src/main/java/me/earth/earthhack/impl/gui/click/component/impl/BooleanComponent.java

```diff
@@ -1,20 +1,19 @@
 package me.earth.earthhack.impl.gui.click.component.impl;
 
 import me.earth.earthhack.api.setting.settings.BooleanSetting;
-import me.earth.earthhack.impl.gui.click.component.Component;
+import me.earth.earthhack.impl.gui.click.component.SettingComponent;
 import me.earth.earthhack.impl.managers.Managers;
 import me.earth.earthhack.impl.util.render.Render2DUtil;
 import me.earth.earthhack.impl.util.render.RenderUtil;
 
-public class BooleanComponent extends Component {
+public class BooleanComponent extends SettingComponent {
```

Comment:
This will give a warning? should it not be like
```java 
public class BooleanComponent extends SettingComponent<Boolean, Setting<Boolean>> {
```

Reply:
When I did this there was no warning. I can change it in about 30 minutes

Reply:
Should be `Raw use of parameterized class 'SettingComponent'`